services/consensusalgo/leanhelixconsensus/test/metrics_test.go

```diff
@@ -0,0 +1,127 @@
+
+		h.beFirstInCommittee() // just so RequestOrderedCommittee will succeed
+		h.expectGossipSendLeanHelixMessage()
+
+		h.handleBlockSync(ctx, 1)
+
+		// Election trigger should fire and metrics should be updated
+
+		metrics := h.getMetrics()
+
+		require.True(t, test.Eventually(1*time.Second, func() bool {
+			return metrics.currentElectionCount.Value() > 0
+		}), "expected currentElectionCount metric to update")
+
+		require.True(t, test.Eventually(1*time.Second, func() bool {
+			return metrics.currentLeaderMemberId.Value() != ""
+		}), "expected currentLeaderMemberId metric to update")
+
+		require.True(t, test.Eventually(1*time.Second, func() bool {
+			matched, err := regexp.MatchString("samples=[1-9]", metrics.timeSinceLastElectionMillis.String())
+			if err != nil {
+				panic(err)
+			}
+			return matched
+		}), "expected timeSinceLastElectionMillis metric to update")
+	})
+
+}
+
+func TestMetricsAreUpdatedOnCommit(t *testing.T) {
+
+	with.Concurrency(t, func(ctx context.Context, parent *with.ConcurrencyHarness) {
+		h := newLeanHelixServiceHarness(0, time.Hour).start(parent, ctx)
+
+		h.expectGossipSendLeanHelixMessage()
+		h.expectValidateTransactionBlock()
+		h.expectValidateResultsBlock()
+		h.expectCommitBlock()
+
+		h.dontBeFirstInCommitee()
+
+		const syncedBlockHeight = 1
+		h.handleBlockSync(ctx, syncedBlockHeight)
+
+		orderedCommittee := h.requestOrderingCommittee(ctx)
+		orderedComitteeNodeIndicies := keys.NodeAddressesForTestsToIndexes(orderedCommittee.NodeAddresses)
+
+		leaderNodeIndex := orderedComitteeNodeIndicies[0] // leader for view 0 is always the first member in the ordered committee
+
+		prevBlockProof := protocol.BlockProofReader(nil) // the previous block (from sync) had no proof
+		randomSeed := randomseed.CalculateRandomSeed(prevBlockProof.RandomSeedSignature())
+
+		const firstTermHeight = syncedBlockHeight + 1
+		const view = 0
+		blockPair := builders.BlockPair().WithHeight(firstTermHeight).WithTransactions(1).WithEmptyLeanHelixBlockProof().Build()
+
+		h.handlePreprepareMessage(ctx, blockPair, firstTermHeight, view, leaderNodeIndex)
+
+		commitMessages := []*interfaces.CommitMessage{}
+		for i := 0; i < h.networkSize(); i++ {
+			cmsg := h.handleCommitMessage(ctx, blockPair, firstTermHeight, view, randomSeed, i)
+			commitMessages = append(commitMessages, cmsg)
+		}
+
+		metrics := h.getMetrics()
+
+		// At this point the first block should be committed, lastCommitTime should update
+		require.True(t, test.Eventually(1*time.Second, func() bool {
+			return metrics.lastCommittedTime.Value() != 0
+		}), "expected lastCommittedTime metric not to update on first commit")
+
+		// timeSinceLastCommitMillis will NOT update because this is the first commit
+		require.True(t, strings.Contains(metrics.timeSinceLastCommitMillis.String(), "samples=0"), "expected lastCommittedTime to not update on first commit")
+
+		// Starting the second term
```

Comment:
is there an easy refactor here to extract the common calls in both terms under a harness/driver method?

Reply:
I preferred not to over complicate the harness for a single test, but maybe we can do something local for this test.

Reply:
extract to a local func